pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -262,3 +271,40 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 	}
 	return nil
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeploymentList *kapi.ReplicationControllerList, deploymentConfig *deployapi.DeploymentConfig) error {
+	// we need to coerce the list type into something we can sort
+	existingDeployments := []*kapi.ReplicationController{}
+	for i := 0; i < len(existingDeploymentList.Items); i++ {
+		existingDeployments = append(existingDeployments, &existingDeploymentList.Items[i])
```

Comment:
It doesn't make sense that this list would contain them. The list is fetched using the client before any action is taken to make a new RC - where would newer deployments be added?